author.tf

```diff
@@ -0,0 +1,428 @@
+  container_name            = "eq-survey-runner-static"
+  container_port            = 80
+  container_tag             = "${var.survey_runner_tag}"
+  application_min_tasks     = "${var.survey_runner_min_tasks}"
+  slack_alert_sns_arn       = "${module.eq-alerting.slack_alert_sns_arn}"
+  alb_listener_path_pattern = "/s/*"
+}
+
+module "author-survey-launcher" {
+  source                 = "github.com/ONSdigital/eq-ecs-deploy?ref=v2.0"
+  env                    = "${var.env}"
+  aws_account_id         = "${var.aws_account_id}"
+  aws_assume_role_arn    = "${var.aws_assume_role_arn}"
+  vpc_id                 = "${module.author-vpc.vpc_id}"
+  dns_zone_name          = "${var.dns_zone_name}"
+  ecs_cluster_name       = "${module.author-eq-ecs.ecs_cluster_name}"
+  aws_alb_arn            = "${module.author-eq-ecs.aws_alb_arn}"
+  aws_alb_listener_arn   = "${module.author-eq-ecs.aws_alb_listener_arn}"
+  service_name           = "author-surveys-launch"
+  listener_rule_priority = 15
+  docker_registry        = "${var.survey_launcher_registry}"
+  container_name         = "go-launch-a-survey"
+  container_port         = 8000
+  healthcheck_path       = "/status"
+  container_tag          = "${var.survey_launcher_tag}"
+  application_min_tasks  = "${var.survey_launcher_min_tasks}"
+  slack_alert_sns_arn    = "${module.eq-alerting.slack_alert_sns_arn}"
+
+  container_environment_variables = <<EOF
+      {
+        "name": "SURVEY_RUNNER_URL",
+        "value": "${module.author-survey-runner.service_address}"
+      },
+      {
+        "name": "SCHEMA_VALIDATOR_URL",
+        "value": "${module.author-schema-validator.service_address}"
+      },
+      {
+        "name": "JWT_ENCRYPTION_KEY_PATH",
+        "value": "${var.survey_launcher_jwt_encryption_key_path}"
+      },
+      {
+        "name": "JWT_SIGNING_KEY_PATH",
+        "value": "${var.survey_launcher_jwt_signing_key_path}"
+      }
+  EOF
+}
+
+module "author" {
+  source                           = "github.com/ONSdigital/eq-ecs-deploy?ref=v2.0"
+  env                              = "${var.env}"
+  aws_account_id                   = "${var.aws_account_id}"
+  aws_assume_role_arn              = "${var.aws_assume_role_arn}"
+  dns_zone_name                    = "${var.dns_zone_name}"
+  ecs_cluster_name                 = "${module.author-eq-ecs.ecs_cluster_name}"
+  vpc_id                           = "${module.author-vpc.vpc_id}"
+  aws_alb_arn                      = "${module.author-eq-ecs.aws_alb_arn}"
+  aws_alb_listener_arn             = "${module.author-eq-ecs.aws_alb_listener_arn}"
+  service_name                     = "author"
+  listener_rule_priority           = 102
+  docker_registry                  = "${var.author_registry}"
+  container_name                   = "eq-author"
+  container_port                   = 3000
+  container_tag                    = "${var.author_tag}"
+  healthcheck_path                 = "/status.json"
+  healthcheck_grace_period_seconds = 60
```

Comment:
Make this 120, as currently author doesn't start in 60 seconds

Reply:
Tom has some thoughts about how we could improve this going forward. It would mean we don't have to build author from scratch each time. Previously he's used placeholder substitution of container env vars into the static HTML at runtime.

We can also use multi stage docker build to strip out all of the node dependencies from the final image and have it serving the static files from nginx instead. He has done it before so we'll look to address that soon, should allow for much faster start up times.